packages/libraries/router/src/persisted_documents.rs

```diff
@@ -0,0 +1,790 @@
+
+            Self { server }
+        }
+
+        fn endpoint(&self) -> String {
+            self.server.url("")
+        }
+
+        /// Registers a valid artifact URL with an actual GraphQL document
+        fn add_valid(&self, document_id: &str) -> Mock {
+            let valid_artifact_url = format!("/apps/{}", str::replace(document_id, "~", "/"));
+            let document = "query { __typename }";
+            let mock = self.server.mock(|when, then| {
+                when.method(GET).path(valid_artifact_url);
+                then.status(200)
+                    .header("content-type", "text/plain")
+                    .body(document);
+            });
+
+            mock
+        }
+    }
+
+    async fn get_body(router_req: router::Request) -> String {
+        let (_parts, body) = router_req.router_request.into_parts();
+        let body = hyper::body::to_bytes(body).await.unwrap();
+        String::from_utf8(body.to_vec()).unwrap()
+    }
+
+    /// Creates a mocked router service that reflects the incoming body
+    /// back to the client.
+    /// We are using this mocked router in order to make sure that the Persisted Documents layer 
+    /// is able to resolve, fetch and pass the document to the next layer.
+    fn create_reflecting_mocked_router() -> MockRouterService {
+        let mut mocked_execution: MockRouterService = MockRouterService::new();
+
+        mocked_execution
+            .expect_call()
+            .times(1)
+            .returning(move |req| {
+                let incoming_body = block_on(get_body(req));
+                Ok(router::Response::fake_builder()
+                    .data(json!({
+                        "incomingBody": incoming_body,
+                    }))
+                    .build()
+                    .unwrap())
+            });
+
+        mocked_execution
+    }
+
+    /// Creates a mocked router service that returns a fake GraphQL response.
+    fn create_dummy_mocked_router() -> MockRouterService {
+        let mut mocked_execution = MockRouterService::new();
+
+        mocked_execution.expect_call().times(1).returning(move |_| {
+            Ok(router::Response::fake_builder()
+                .data(json!({
+                    "__typename": "Query"
+                }))
+                .build()
+                .unwrap())
+        });
+
+        mocked_execution
+    }
+
+    #[tokio::test]
+    async fn should_allow_arbitrary_when_regular_req_is_sent() {
+        let service = create_reflecting_mocked_router();
+        let service_stack = PersistedDocumentsPlugin::new(Config {
+            enabled: Some(true),
+            endpoint: Some("https://cdn.example.com".into()),
+            key: Some("123".into()),
+            allow_arbitrary_documents: Some(true),
+            connect_timeout: Default::default(),
+            request_timeout: Default::default(),
+            accept_invalid_certs: Default::default(),
+            cache_size: Default::default(),
```

Comment:
simplify with:

```rust
..Default::default()
```

to add value to the rest of the things that need a default value.